Support for long paths in Windows

Awscar opened the pull request:
When dealing with deep folder trees and/or long file paths, MAX_PATH can get too small to contain them.
For these cases, Windows provides UNC paths, which are absolute paths prefixed with the "\\\\?\\" string for local scope, and "\\\\?\\UNC\\" for network.

For example, with the previous ZPAQ version, the next experimental file won't be compressed because the file scanning engine can't reach it, will be silently skipped, getting your backup compromised:

C:\Users\Oscar\Documents\Python\Develop\ZPAQ\sample1\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\long_file_path.txt

With UNC support it gets reached and compressed.

Awscar committed:
Support for long paths in Windows

When dealing with deep folder trees and/or long file paths, MAX_PATH can get too small to contain them.
For these cases, Windows provides UNC paths, which are absolute paths prefixed with the "\\\\?\\" string for local scope, and "\\\\?\\UNC\" for network.

For example, with the previous ZPAQ version, the next experimental file won't be compressed because the file scanning engine can't reach it:

C:\Users\Oscar\Documents\Python\Develop\ZPAQ\sample1\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\0123456789\long_file_path.txt

With UNC support it gets reached and compressed.

Awscar committed:
Update zpaq.cpp

Windows: 
Ensure that paths don't include "/" char, just "\\", they may fail when accessing shared folders (not tested in all Windows versions).

For example, these commands may always work:
	dir "\\server\share\folder"
	dir "\\server/share\folder"
	dir "\\server\share\"
	dir "\\server\share\*"

However these may not:
	dir "\\server\share/*"
	dir "\\server/share\*"
	dir "\\server/share/*"

szepeviktor commented:
Hello @Awscar!
This code doesn't reach Matt. He doesn't use GitHub, this is just a mirror.

Awscar committed:
Added "-reparseCopy" switch to admit reparse files

Symbolic links to files are by default discarded. With this switch, they are treated as *normal* files and copied (the link disappears).
Note that this means there will be duplicated files when restored.
For example, if your source file tree had *two links* to *one file* before backup, when restored there will be *two files* (identical) and *no link*.

Awscar commented:
> 
Thanks @szepeviktor

Do you mean he will not validate/merge any pull requests, even in this mirror?
Is there another repository to propose changes other than this?

Anyway, as my only idea is to place my commits in a place to be useful to others, and honor the great tool that is zpaq to me, this mirror can be a valid place, unless there is a better one.

szepeviktor commented:
> Do you mean he will not

He will not **see** it!!

szepeviktor commented:
> Is there another repository to propose changes other than this?

Try https://encode.su/threads/456-zpaq-updates/page90

lygstate commented:
Does Matt still working on this, As it's a long time that no new version are released

fcorbelli commented:
After a year (!) my suggestion
Post on zpaqfranz, my own fork

Neustradamus commented:
Dear @lygstate,

Any progress on this @Awscar PR?

Can you add "Issues" section?

Thanks in advance.

lygstate commented:
this is just a mirror, but I can create a development branch for deal with issues， as the author did not publish any new version for a long time
